modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/segstore/RaftLogCheckpointer.java

```diff
@@ -137,7 +137,8 @@ long firstLogIndexInclusive(long groupId) {
         while (it.hasNext()) {
             SegmentInfo segmentInfo = it.next().memTable().segmentInfo(groupId);
```

Comment:
I believe I'm asking the same question second time here. Is it possible that a constant need to resolve `groupId` into various structures will affect the performance?

Reply:
I wonder why you iterate over multiple memtables in this method. I thought that a single checkpoint only corresponds to a single memtable, is this assumption false?

Reply:
> and we search inside the queue during reads.

I did not realize that this is a method for reading the data, not for internal usage of the checkpointer. No more questions